aws-rds-dbparametergroup/src/main/java/software/amazon/rds/dbparametergroup/BaseHandlerStd.java

```diff
@@ -135,13 +135,17 @@ protected ProgressEvent<ResourceModel, CallbackContext> applyParametersWithReset
         if (callbackContext.isParametersApplied())
             return ProgressEvent.defaultInProgressHandler(callbackContext, NO_CALLBACK_DELAY, model);
         //Maps will be populated in upcoming calls in progress chain.
-        final Map<String, Parameter> defaultEngineParameters = Maps.newHashMap();
-        final Map<String, Parameter> currentDBParameters = Maps.newHashMap();
+        final Map<String, Parameter> defaultEngineParameters = callbackContext.getDefaultParameters();
+        final Map<String, Parameter> currentDBParameters = callbackContext.getCurrentParameters();
 
         return ProgressEvent.progress(model, callbackContext)
-                .then(progressEvent -> describeDefaultEngineParameters(progressEvent, defaultEngineParameters, proxy, proxyClient, null, requestLogger))
+                .then(progressEvent -> Commons.execAndSaveContextIfConditionIsNotMet(progressEvent,
+                        () -> describeDefaultEngineParameters(progressEvent, defaultEngineParameters, proxy, proxyClient, callbackContext.getDefaultParametersMarker(), requestLogger),
+                        CallbackContext::isDefaultParametersFetched))
```

Comment:
As describeDefaultEngineParameters alters the context already, we can blackbox the exec-once approach completely. This will eliminate the need for change here and below.